result/src/commonMain/kotlin/com/github/kittinunf/result/Result.kt

```diff
@@ -110,6 +110,11 @@ inline fun <V, reified E : Throwable> List<Result<V, E>>.lift(): Result<List<V>,
         }
     }
 
+inline fun <V, reified E : Throwable> List<Result<V, E>>.liftSuccess(doWithErrors: (e: List<E>) -> Unit): Result<List<V>, E> {
```

Comment:
I wonder whether we should just add `doWithErrors` into the regular lift() and default as do nothing? What do you think? 🤔

Reply:
I thought this too, but If I understand the ask in the issue, the idea is that it is a similar functionality of lift, but it will return a Result.success containing success V even if there is 1 or more errors which would be a change to the current behavior. I was messing around with a similar implementation that is sort of inline with what you are thinking.

```kotlin
inline fun <V, reified E : Throwable> List<Result<V, E>>.lift(): Result<List<V>, E> = liftResult { successes, errors ->
    when (errors.isEmpty()) {
        true -> Result.success(successes)
        else -> Result.failure(errors.first())
    }
}

inline fun <V, reified E : Throwable> List<Result<V, E>>.liftResult(fn: (v: List<V>, e: List<E>) -> Result<List<V>, E>): Result<List<V>, E> =
    fold(Pair<MutableList<V>, MutableList<E>>(mutableListOf(), mutableListOf())) { acc, result ->
        result.success { acc.first.add(it) }
        result.failure { acc.second.add(it) }
        acc
    }.let { fn(it.first, it.second) }

```

This preserves the current behavior of `lift` (unit tests all pass) while also allowing the caller to use a method that handles what they want the Result to be based on 1 or more errors or successes. and what to do with the errors rather than throwing them out. If you like it, let me know and I can update the PR, or I can add the doWithErrors to lift and leave the existing functionality 👍

Reply:
I like this 👏 !

Reply:
what do you think about using the simple as lift() though? like 

```kotlin
`lift()` --> essentially switch to error if any errors has something

`lift(fn: (v: List<V>, e: List<E>) -> Result<List<V>, E>)` --> if user wish to do something else with the List of success Result or Failure ...
```

Also, maybe we wanna name the parameter better, it looks like a `policy` or `condition` something along that line ...

Reply:
@kittinunf I made some naming changing. I think keeping it called just lift and overloading it makes more sense too. Let me know if you have any more thoughts and thank you for your time! 🙏 😊